tembo-cli/src/cli/docker.rs

```diff
@@ -41,24 +42,57 @@ impl Docker {
     }
 
     // Build & run docker image
-    pub fn build_run(instance_name: String) -> Result {
+    pub fn build_run(instance_name: String) -> Result<i32> {
         let mut sp = Spinner::new(Spinners::Line, "Running Docker Build & Run".into());
+        let container_list = Self::container_list_filtered(&instance_name).unwrap();
+
+        if container_list.contains(&instance_name) {
+            let container_port = Docker::get_container_port(container_list)?;
 
-        if Self::container_list_filtered(&instance_name)
-            .unwrap()
-            .contains(&instance_name)
-        {
             sp.stop_with_message("- Existing container found".to_string());
+            Ok(container_port)
         } else {
+            let port = Docker::get_available_port()?;
+
             let command = format!(
-                "docker build . -t postgres && docker run --name {} -p 5432:5432 -d postgres",
-                instance_name
+                "docker build . -t postgres && docker run --name {} -p {}:{} -d postgres",
+                instance_name, port, port
             );
             run_command(&command)?;
             sp.stop_with_message("- Docker Build & Run completed".to_string());
+            Ok(port)
         }
+    }
 
-        Ok(())
+    fn get_available_port() -> Result<i32> {
+        let ls_command = "docker ps -a --format '{{.Ports}}'";
```

Comment:
I'm wondering if we really need the `-a`

Reply:
Good catch. Not needed. Updated it.